### PR TITLE
Misc cleanup

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

File: storage/store.go [hunk 1]
```diff
@@ -1324,8 +1319,7 @@ func (s *Store) ProposeRaftCommand(idKey cmdIDKey, cmd proto.InternalRaftCommand
 		panic("proposed a nil command")
 	}
 	// Lazily create group. TODO(bdarnell): make this non-lazy
-	err := s.multiraft.CreateGroup(uint64(cmd.RaftID))
-	if err != nil {
+	if err := s.multiraft.CreateGroup(uint64(cmd.RaftID)); err != nil {
```
Comment (tschottdorf) [inline storage/store.go]: I thought we had a rule about not changing unrelated code for personal preference. I don't want to speak for Ben but I think I remember he preferred the original form in cases like these.

Comment (tamird) [inline storage/store.go]: My understand of the rule was to not change those things in the same commit as other, orthogonal changes. In this case, I think this is cleaner because it doesn't declare `err` for the code that follows, and this line is short.

File: storage/range.go [hunk 2]
```diff
@@ -433,7 +433,7 @@ func (r *Range) Desc() *proto.RangeDescriptor {
 // SetDesc atomically sets the range's descriptor. This method calls
 // processRangeDescriptorUpdate() to make the range manager handle the
 // descriptor update. Note that processRangeDescriptorUpdate()
-// acquires the metaLock.
+// acquires a lock on the range manager.
```
Comment (tschottdorf) [inline storage/range.go]: it's not a lock on the range manager (last time I checked). I agree `metaLock` doesn't really tell you anything though, not even which mutex it is which is being held. Or is it not talking about the MetaLock? I'm confused.
Also, isn't the next line in the comment contradicting the previous?

Comment (tamird) [inline storage/range.go]: So, `metaLock` is actually a member of `Range`, and it is explicitly locked in `AdminSplit`, `AdminMerge`, and `ChangeReplicas`.

Actually, we should remove this line, since the lock that `processRangeDescriptorUpdate` acquires when `rm` is a `Store` happens to acquire the `Store`'s `mu`, but that's not generally true for implementers of `rangeManager`.

File: storage/store.go (hunk 1 above)
Comment (tschottdorf) [inline storage/store.go]: I'm not going to argue for one or the other, but my understanding is that this is a change based on your personal preference for less local scope in exchange for longer lines. I generally prefer to leave things alone unless you feel strongly about something. Correct me if you think otherwise, but this seems like a nit.

Comment (tamird) [inline storage/store.go]: Rescinded.

**Comment by tschottdorf** [comment]
can you add a TODO in split_queue.go:107 about why it's not just calling AdminSplit as in the case immediately above?

File: storage/range.go [hunk 3]
```diff
@@ -430,15 +430,14 @@ func (r *Range) Desc() *proto.RangeDescriptor {
...
 }
 
-// SetDesc atomically sets the range's descriptor. This method calls
+// setDesc atomically sets the range's descriptor. This method calls
 // processRangeDescriptorUpdate() to make the range manager handle the
-// descriptor update. Note that processRangeDescriptorUpdate()
-// acquires the metaLock.
+// descriptor update.
 //
 // This method should be called in the context of having metaLock held,
```
Comment (tschottdorf) [inline storage/range.go]: Looks like this comment can go? The TODO could be moved next to the metaLock field in Range.

**Comment by tschottdorf** [comment]
LGTM
